MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -23,27 +23,43 @@
 
 using namespace std;
 
+// Some Metal formats are not supported by the headers on certain platforms. However, formats have
+// been 'unlocked' on some platforms with newer versions of Xcode.
 
 // Add stub defs for unsupported MTLPixelFormats per platform
 #if MVK_MACOS
-#   if !MVK_MACOS_APPLE_SILICON
+#	if !MVK_XCODE_12_2 // macOS 11.0 / iOS 14.2
```

Comment:
Have we tested this on `x86_64` hardware? The formats are defined in macOS 11 and above, but can they actually be used on non-Apple Silicon macOS devices?

And I'm not seeing how this is fixing the original issue, which was to expose formats like `R4G4B4A4_PACK16` for `x86_64` when running on Rosetta (as an Apple Silicon build already seems to be covered in `modifyMTLFormatCapabilities()`). 

Am I missing something?

Reply:
>>> but can they actually be used on non-Apple Silicon macOS devices?

Yes, there's no distinction between arm64 and x86_64 targets. It is purely up to the driver to expose the features and device family. The Metal driver for the M1 will be the same no matter if its arm64 or x86_64.

>>> And I'm not seeing how this is fixing the original issue

The SDK builds the drivers for x86_64 because then the binaries will run on both x86_64 and arm64. Which means that the build target is not x86_64. `MVK_MACOS_APPLE_SILICON` is only `1` when the target is `arm64`. So simply by replacing the macro with a check for the SDK versions (`MVK_XCODE_12_2`) we also include any binary of MoltenVK built for x86_64 that could possibly support the GPU families and therefore also the missing formats. In short, the code was wrongfully assuming that all the formats/families are only available for arm64 builds. I opened the issue because I witnessed people reporting the issue because they were using the SDK build of MoltenVK and I only looked at other GPUInfo reports that also used that build. I originally had thought there were missing/wrong checks for the formats but I was wrong.

Reply:
> The SDK builds the drivers for x86_64 because then the binaries will run on both x86_64 and arm64.

As an aside, MoltenVK and the ICD loader really need to be built universal `arm64` + `x86_64`; otherwise, native `arm64` binaries can't link to Vulkan or use MoltenVK. But that's an SDK issue.

Reply:
Ah. Okay. I think I see how you're piecing this together now. 

You're just exposing the formats for all macOS builds, and the actual format capability enabling is enabled by `modifyMTLFormatCapabilities()` at runtime:

	addGPUOSMTLPixFmtCaps( Apple5, 10.16, ABGR4Unorm, RFCMRB );

But that is also guarded with a  `#if MVK_MACOS_APPLE_SILICON` macro, so you'll have to reorganize that area too, won't you?

Reply:
Yeah, I must've overlooked that. I will overlook the build tomorrow on a specific x86_64 build on my machine so I can properly test how and if this works in all cases.

Reply:
Had to fiddle with Xcode a bit but I got MoltenVK to build a binary which only contained x86_64 code. With that change I did manage to see how all the formats disappeared (just like a SDK build). I also tried to run a Khronos Vulkan Sample (with some texture changed to R4G4B4A4) but the loader didn't like that the dylib was compiled exclusively for x86_64, even though I do have Rosetta 2 installed. Though this should definitely work.